### ISSUE TITLE
Netty4ClientHttpRequestFactory is Deprecated [suggestion]

### ISSUE THREAD

**Issue by zhangxianwei2015** [thingsboard/thingsboard#3124]
https://github.com/thingsboard/thingsboard/blob/fc597f2c0bc25bbcf39f72f516ef8d4cd5a27f3c/rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/rest/TbHttpClient.java#L135

Try send the http request by TbRestApiCallNode, when raise to 1w/s, it begin to lag and loss.

I change to use okhttpclient it working fine.

**Comment by yefimov-andrey** [comment]
What exactly are you suggesting to change to okhttpclient?

**Comment by zhangxianwei2015** [comment]
Hi @yefimov-andrey ,
I am using thingboard2.4.3 CE.
The case is I was doing a testing, with 10000 clients connect to thingsboard and send 20000 telemetry messages per second. I try to use the "Rest Api Call" node to forwards these messages to third apps In the rule chain, but I found the messages were blocked in the "Rest Api Call" node, http requests are lag and loss.

I try to create several threads in the node, each have a http client to process the messages.
I first tried to push the msg to a queue when received by akka,
    @Override
    public void onMsg(TbContext ctx, TbMsg msg) {
        blockingQueue.offer(msg);
    }

then create several threads to take message from the queue and process the message by http client.
the output increased to 5000messages/s, but in the thead stack there are a lot of theads in epollWait(I don't know the root cause),

 java.lang.Thread.State: RUNNABLE
        at sun.nio.ch.EPollArrayWrapper.epollWait(Native Method)
        at sun.nio.ch.EPollArrayWrapper.poll(EPollArrayWrapper.java:269)
        at sun.nio.ch.EPollSelectorImpl.doSelect(EPollSelectorImpl.java:93)
        at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)

As I notice AsyncRestTemplate in springboot is Deprecated, so I tried to use OkHttpClient and the output increase to more than 2w/s.

I am not sure if I use the "rest api call" correctly, it was just a testing, and in the case the OkHttpClient is perform better.
